File: Sources/Internal/ATTNAppInfo.m
//
//  AppInfo.m
//  attentive-ios-sdk-framework
//
//  Created by Wyatt Davis on 3/14/23.
//

#import <Foundation/Foundation.h>
#if TARGET_OS_IOS
#import <UIKit/UIKit.h>
#endif

#import "ATTNAppInfo.h"
#import "ATTNVersion.h"

@implementation ATTNAppInfo

+ (NSString *)getAppBuild {
  return [[NSBundle mainBundle] infoDictionary][@"CFBundleVersion"];
}
+ (NSString *)getAppVersion {
  return [[NSBundle mainBundle] infoDictionary][@"CFBundleShortVersionString"];
}
+ (NSString *)getAppName {
  return [[NSBundle mainBundle] infoDictionary][@"CFBundleName"];
}
+ (NSString *)getAppId {
  return [[NSBundle mainBundle] bundleIdentifier];
}

+ (NSString *)getDeviceModelName {
  #if TARGET_OS_IOS
    return [[UIDevice currentDevice] model];
  #else
    return @"Not iOS device, not supported";
  #endif
}

+ (NSString *)getDevicePlatform {
  #if TARGET_OS_IOS
    return [[UIDevice currentDevice] systemName];
  #else
    return @"Not iOS device, not supported";
  #endif
}

+ (NSString *)getDeviceOsVersion {
  return [[NSProcessInfo processInfo] operatingSystemVersionString];
}

+ (NSString *)getSdkName {
  return @"attentive-ios-sdk";
}

+ (NSString *)getSdkVersion {
  return SDK_VERSION;
}

@end

File: Sources/Internal/ATTNAppInfo.swift
//
//  ATTNAppInfo.swift
//  attentive-ios-sdk-framework
//
//  Created by Vladimir - Work on 2024-05-31.
//

import Foundation
import UIKit

open class ATTNAppInfo {
  open class func getAppBuild() -> String {
    Bundle.main.object(forInfoDictionaryKey: "CFBundleVersion") as? String ?? ""
  }

  open class func getAppVersion() -> String {
    Bundle.main.object(forInfoDictionaryKey: "CFBundleShortVersionString") as? String ?? ""
  }

  open class func getAppName() -> String {
    Bundle.main.object(forInfoDictionaryKey: "CFBundleName") as? String ?? ""
  }

  open class func getAppId() -> String {
    Bundle.main.bundleIdentifier ?? ""
  }

  open class func getDeviceModelName() -> String {
    UIDevice.current.model
  }

  open class func getDevicePlatform() -> String {
    UIDevice.current.systemName
  }

  open class func getDeviceOsVersion() -> String {
    ProcessInfo.processInfo.operatingSystemVersionString
  }

  open class func getSdkName() -> String {
    "attentive-ios-sdk"
  }

  open class func getSdkVersion() -> String {
    ATTNConstants.sdkVersion
  }
}

public extension ATTNAppInfo {
  static func getFormattedAppName() -> String {
    getAppName().replacingOccurrences(of: " ", with: "-")
  }
}
